features/earn/aave/open/sidebars/GetSidebarTexts.tsx

```diff
@@ -0,0 +1,58 @@
+import { Event, EventData, SingleOrArray } from 'xstate'
+
+import { SidebarSectionFooterButtonSettings } from '../../../../../components/sidebar/SidebarSectionFooter'
+import { getProxyButtonSettings } from '../../../../proxyNew/getProxyButtonSettings'
+import { OpenAaveEvent, OpenAaveStateMachineState } from '../state/openAaveStateMachine.types'
+
+export function GetSidebarTexts(
+  state: OpenAaveStateMachineState,
+  t: (key: string, options?: any) => string,
+  send: (
+    event: SingleOrArray<Event<OpenAaveEvent>>,
+    payload?: EventData | undefined,
+  ) => OpenAaveStateMachineState,
+): SidebarSectionFooterButtonSettings {
+  const isSuccessStage = state.matches('txSuccess')
+  const isLoading = state.matches('txInProgress')
+
+  const url = isSuccessStage ? `/${state.context.vaultNumber}` : undefined
+
+  const steps: [number, number] | undefined = !isSuccessStage
+    ? [state.context.currentStep!, state.context.totalSteps!]
+    : undefined
+
+  const canCreateProxy = state.can('CREATE_PROXY')
+
+  const baseSettings = {
+    steps,
+    isLoading,
+    url,
+    disabled: !state.context.canGoToNext,
+  }
```

Comment:
I found `canGoToNext` slightly confusing just because we have transition guards. Appreciate this is for use in disabling button.

What do you think about following maybe?


```suggestion
  const canCreateProxy = state.can(states.CREATE_PROXY)
  const canConfirmDeposit = state.can(states.CONFIRM_DEPOSIT)
  const canProceed = canCreateProxy || canConfirmDeposit

  const baseSettings = {
    steps,
    isLoading,
    url,
    disabled: !canProceed,
  }
```